niceman/distributions/vcs.py

```diff
@@ -335,6 +336,15 @@ def hexsha(self):
             # might still be the first yet to be committed state in the branch
             return None
         
+    @property
+    def root_hexsha(self):
+        try:
+            rev_list = self._run_git('rev-list master').split('\n')
```

Comment:
note that `master` might not be available... for later may be

Reply:
Sorry about the `--reverse -n 1` misinformation.

From `man git-rev-list`:

```
Note that these are applied before commit ordering and formatting options, such as --reverse.

 -<number>, -n <number>, --max-count=<number>
     Limit the number of commits to output.
```

`--reverse -n 1` reduces the revision list to the latest revision and then reverses the list.  Obviously not what we want.

Reply:
We could instead use:

```
git rev-list --max-parents=0 master
```

Give the commit with 0 parents, i.e., the root commit.

Reply:
> checked for additional options to see if there is a way to limit to the left one... but for some reason it is not working as I thought it would, ie. I guess they do not apply here

`--{left,right}-only` aren't about parents; they're about symmetric differences (i.e., `rev-list A...B`).  You could use `--first-parent` to get the first parent.

> we could replace master with HEAD -- we care about current state, so it would be only reasonable imho

I agree.